src/lib.rs

```diff
@@ -114,6 +114,38 @@ pub fn hash(bytes: &[u8]) -> HashMatrix {
     acc
 }
 
+#[must_use]
+#[inline]
+#[cfg(feature = "std")]
+/// Same as `hash` but computes the hash of the byte stream in parallel.
+///
+/// The number of threads used is set automatically but can be overridden using the
+/// `RAYON_NUM_THREADS` environment variable
+pub fn hash_par(bytes: &[u8]) -> HashMatrix {
+    use rayon::prelude::*;
+
+    bytes
+        .par_chunks(2)
+        .fold(
+            || I,
+            |mut acc, bs| {
```

Comment:
Only nit with this PR: Rather than mutating `acc`, you could just return the result of the expression. At least on my machine, there's no noticeable performance penalty for this change.